packages/fx-core/src/plugins/resource/bot/functionsHostedBot/plugin.ts

```diff
@@ -109,4 +126,105 @@ export class FunctionsHostedBotImpl extends TeamsBotImpl {
+    );
+
+    Logger.info(Messages.DeployingBot);
+
+    const workingDir = this.config.scaffold.workingDir;
+    if (!workingDir) {
+      throw new PreconditionError(Messages.WorkingDirIsMissing, []);
+    }
+
+    const programmingLanguage = this.config.scaffold.programmingLanguage;
+    if (!programmingLanguage) {
+      throw new PreconditionError(Messages.SomethingIsMissing(PluginBot.PROGRAMMING_LANGUAGE), []);
+    }
+
+    const deployTime: Date = new Date();
+    const deployMgr = new FuncHostedDeployMgr(workingDir, this.ctx.envInfo.envName);
+    const needsToRedeploy: boolean = await deployMgr.needsToRedeploy([
+      FolderNames.NODE_MODULES,
+      ...(await deployMgr.getIgnoreRules(FuncHostedBotDeployConfigs.FUNC_IGNORE_FILE)),
+      ...(await deployMgr.getIgnoreRules(FuncHostedBotDeployConfigs.GIT_IGNORE_FILE)),
+    ]);
+    if (!needsToRedeploy) {
+      Logger.debug(Messages.SkipDeployNoUpdates);
+      return ResultFactory.Success();
+    }
+
+    const handler = await ProgressBarFactory.newProgressBar(
+      ProgressBarConstants.DEPLOY_TITLE,
+      ProgressBarConstants.DEPLOY_STEPS_NUM,
+      this.ctx
+    );
+
+    await handler?.start(ProgressBarConstants.DEPLOY_STEP_START);
+
+    await handler?.next(ProgressBarConstants.DEPLOY_STEP_NPM_INSTALL);
+    await LanguageStrategy.localBuild(programmingLanguage, workingDir);
+
+    await handler?.next(ProgressBarConstants.DEPLOY_STEP_ZIP_FOLDER);
+
+    const rules = await deployMgr.getIgnoreRules(FuncHostedBotDeployConfigs.FUNC_IGNORE_FILE);
+    const zipBuffer = await deployMgr.zipAFolder(rules);
+
+    // 2.2 Retrieve publishing credentials.
+    const webSiteMgmtClient = new appService.WebSiteManagementClient(
+      await this.getAzureAccountCredential(),
+      this.config.provision.subscriptionId!
+    );
+    const listResponse = await AzureOperations.ListPublishingCredentials(
+      webSiteMgmtClient,
+      this.config.provision.resourceGroup!,
+      this.config.provision.siteName!
+    );
+
+    const publishingUserName = listResponse.publishingUserName
+      ? listResponse.publishingUserName
+      : "";
+    const publishingPassword = listResponse.publishingPassword
+      ? listResponse.publishingPassword
+      : "";
+    const encryptedCreds: string = utils.toBase64(`${publishingUserName}:${publishingPassword}`);
+
+    const config = {
+      headers: {
+        "Content-Type": "application/octet-stream",
+        "Cache-Control": "no-cache",
+        Authorization: `Basic ${encryptedCreds}`,
+      },
+      maxContentLength: Infinity,
+      maxBodyLength: Infinity,
+      timeout: CommonConstants.deployTimeoutInMs,
+    };
+
+    const zipDeployEndpoint: string = getZipDeployEndpoint(this.config.provision.siteName!);
+    await handler?.next(ProgressBarConstants.DEPLOY_STEP_ZIP_DEPLOY);
+    await AzureOperations.ZipDeployPackage(zipDeployEndpoint, zipBuffer, config);
+
+    await AzureOperations.RestartWebApp(
```

Comment:
Why need restart?

Reply:
As I discussed with @hund030 offline, the function plugin has met some problems and the solution is to restart.